### PR TITLE
2.6.0/sdk improvements part 1

### PR DESCRIPTION
**Pull request by MusiikoAS**
Removed isS2S parameter 
Deprecate language parameter 
Separate setSessionId and set dataOptions into 2 different functions

### CODE REVIEW

File: src/InBrain.tsx [hunk 1]
```diff
@@ -157,6 +173,8 @@ const optionsActions: { [key in StylingOptionName]: ((params: any) => Promise<an
 export default {
     init,
     showSurveys,
+    setSessionID,
+    setDataOptions,
```
Comment (susslik) [inline src/InBrain.tsx:177]: Remove setDataOptions from here if possible (I'm not sure what is that, if this is just list of available methods - no problem, let it be then)

File: src/Options.tsx [hunk 2]
```diff
@@ -8,6 +8,9 @@ export type StatusBarOptionName = "lightStatusBar";
 
 // *** CONFIG OPTION TYPE
 export type ConfigOptionTypes = {
```
Comment (susslik) [inline src/Options.tsx:10]: deprecate dataPoints here as well

**Review by susslik — CHANGES_REQUESTED** [review]

File: ios/InBrainSurveys.m [hunk 3]
```diff
@@ -118,14 +120,23 @@ + (BOOL)requiresMainQueueSetup
...
+RCT_EXPORT_METHOD(setInBrain:(NSString *)apiClientId apiSecret:(nonnull NSString *)apiSecret userId:(nonnull NSString *)userId resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
 {
     @try {
 
         [self notNull:@"apiClientId" toCheck:apiClientId];
         [self notNull:@"apiSecret" toCheck:apiSecret];
         [self notNull:@"userId" toCheck:userId];
 
+        //hardcoded isS2S (always True)
+        bool isS2S = true;
```
[the remark this answers is not in the document]
Comment (MusiikoAS) [reply ios/InBrainSurveys.m:132]: I'm not sure about this, I believe we must indicate some how, this param

File: android/src/main/java/com/inbrain/InBrainSurveysModule.java [hunk 4]
```diff
@@ -102,6 +119,46 @@ public void setParam(HashMap<String, String> param) {
...
+    // ***** SET INBRAIN SESSION ID *****
+    // **********************************
+    @ReactMethod
+    public void setSessionID(final String sessionId, Promise promise) {
+        //tmp data until android sdk new version
+        this.sessionID = sessionId;
+        HashMap<String, String> data = this.sessionData;
+
+        //tmp check until android sdk new version
+        if(data != null) {
```
[the remark this answers is not in the document]
Comment (MusiikoAS) [reply android/src/main/java/com/inbrain/InBrainSurveysModule.java]: done

File: android/src/main/java/com/inbrain/InBrainSurveysModule.java [hunk 5]
```diff
@@ -102,6 +119,46 @@ public void setParam(HashMap<String, String> param) {
...
+    // ***** SET INBRAIN DATA POINTS *****
+    // **********************************
+    @ReactMethod
+    public void setDataOptions(final ReadableMap data, Promise promise) {
+        //tmp data until android sdk new version
+        this.sessionData = toHashMap(data);
+        String sessionId = this.sessionID;
+
+        //tmp check until android sdk new version
+        if(sessionId != null) {
```
[the remark this answers is not in the document]
Comment (MusiikoAS) [reply android/src/main/java/com/inbrain/InBrainSurveysModule.java]: done

File: android/src/main/java/com/inbrain/InBrainSurveysModule.java [hunk 6]
```diff
@@ -55,13 +60,16 @@ public String getName() {
...
-    public void setInBrain(final String apiClientId, final String clientSecret, final boolean isS2S, final String userId, final Promise promise) {
+    public void setInBrain(final String apiClientId, final String clientSecret, final String userId, final Promise promise) {
         try {
             // Validate parameters
             notNull("apiClientId", apiClientId);
             notNull("clientSecret", clientSecret);
             notNull("userId", userId);
 
+            //hardcoded isS2S (always True)
+            Boolean isS2S = true;
```
[the remark this answers is not in the document]
Comment (MusiikoAS) [reply android/src/main/java/com/inbrain/InBrainSurveysModule.java:71]: I'm not sure about this, I believe we must indicate some how, this param

File: android/src/main/java/com/inbrain/InBrainSurveysModule.java [hunk 7]
```diff
@@ -102,6 +109,28 @@ public void setParam(HashMap<String, String> param) {
 
     }
 
+    // **********************************
+    // ***** SET INBRAIN SESSION ID *****
+    // **********************************
+    @ReactMethod
+    public void setSessionID(final String sessionId, Promise promise) {
```
[the remark this answers is not in the document]
Comment (MusiikoAS) [reply android/src/main/java/com/inbrain/InBrainSurveysModule.java:116]: done

File: android/src/main/java/com/inbrain/InBrainSurveysModule.java [hunk 8]
```diff
@@ -102,6 +109,28 @@ public void setParam(HashMap<String, String> param) {
...
+        this.sessionID = sessionId;
+        HashMap<String, String> data = this.sessionData;
+        InBrain.getInstance().setInBrainValuesFor(sessionId, data);
+    }
+
+    // **********************************
+    // ***** SET INBRAIN DATA POINTS *****
+    // **********************************
+    @ReactMethod
+    public void setDataOptions(final ReadableMap data, Promise promise) {
```
[the remark this answers is not in the document]
Comment (MusiikoAS) [reply android/src/main/java/com/inbrain/InBrainSurveysModule.java:127]: done

File: ios/InBrainSurveys.m [hunk 9]
```diff
@@ -137,31 +148,31 @@ + (BOOL)requiresMainQueueSetup
...
+    [self.inbrain setSessionID:sessionUId];
+}
 
-        [keys enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop) {
-            [mapped addObject:@{ obj : [data objectForKey:obj]}];
-        }];
+// **********************************
+// ***** SET INBRAIN DATA POINTS *****
+// **********************************
+RCT_EXPORT_METHOD(setDataOptions:(NSDictionary *)data resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
```
[the remark this answers is not in the document]
Comment (MusiikoAS) [reply ios/InBrainSurveys.m:161]: done

File: src/InBrain.tsx (hunk 1 above)
Comment (MusiikoAS) [reply src/InBrain.tsx:177]: yes, it's listing of available methods

File: src/Options.tsx (hunk 2 above)
Comment (MusiikoAS) [reply src/Options.tsx:10]: done

File: ios/InBrainSurveys.m (hunk 3 above)
[the remark this answers is not in the document]
Comment (susslik) [reply ios/InBrainSurveys.m:132]: Removed

File: android/src/main/java/com/inbrain/InBrainSurveysModule.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (susslik) [reply android/src/main/java/com/inbrain/InBrainSurveysModule.java:71]: Removed

**Review by susslik — APPROVED** [review]
Hey @MusiikoAS - the updates looks great!

**Pull request merged**